### PR TITLE
Clear selecteds if selectables do not contain an event target

### LINKED ISSUE #2091: [BUG] UiText cursor moves when clicking outside the transform
## Description

A selected editable text field seems to be affected by clicks outside of its transform, and can't be deselected.

## Reproduction Steps

1. `cargo run --example ui --features vulkan`
2. Click inside the "Editable" text transform to select it.
3. Click anywhere else on the screen.

## What You Expected to Happen

It should be deselected.

## What Actually Happened

The cursor moves to the closest glyph at the X position that clicked, instead of being deselected. This should only happen if the click was inside the bounds of the transform.

## Your Environment

1. Operating System: Void Linux
2. Operating System Version: latest
3. Version of Amethyst: tested on master, 0.13, 0.12, and 0.11.

### PR DESCRIPTION
**Pull request by mvlabat**
## Description

An attempt to fix #2091

I'm not sure what this comment is about:
https://github.com/amethyst/amethyst/blob/72f7d8f913b4a4d8f05733f2d0c27315d6a09530/amethyst_ui/src/selection.rs#L257
So maybe my change breaks something..

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Updated the content of the book if this PR would make the book outdated.
- [ ] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [ ] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [ ] Ran `cargo clippy --all --features "empty"`
- [ ] Ran `cargo test --all --features "empty"`

### CODE REVIEW

**Comment by mvlabat** [comment]
Also the caret doesn't disappear at once, one can notice that it moves right before disappearing. And if we click on the text field again, it highlights (selects) some characters, where the start position was the previous character to which the caret moved right before disappearing.

Maybe that's something related to what is described here:
https://github.com/amethyst/amethyst/issues/2146#issuecomment-590244355

**Comment by azriel91** [comment]
Cool, the code fixes half the issue (deselects the editable widget when clicking outside), but it deselects it after it responds to it the first time.

> And if we click on the text field again, it highlights (selects) some characters, where the start position was the previous character to which the caret moved right before disappearing.

I can't seem to reproduce the highlighting behaviour though.

Shall try and fix the "move to the correct position on click" and the "incorrectly respond to first click outside the box" issues now.

**Comment by mvlabat** [comment]
*(cross-posting from Discord)*

Ok, I think I've fixed it. The thing I don't like is that we use `InputHandler` in `amethyst_ui/src/selection.rs:240`. That makes it dependent on `InputSystem`, which I had to specify as a dependency in `UiBundle`. It's a breaking change (well, I think it's not a big deal on its own), but it also can be quite obscure to users: the application will def crash if `InputBundle` is loaded after `UiBundle`

File: amethyst_ui/src/text.rs [hunk 1]
```diff
@@ -194,74 +194,82 @@ impl<'a> System<'a> for TextEditingMouseSystem {
...
+                    event:
+                        WindowEvent::MouseInput {
+                            button: MouseButton::Left,
+                            state,
+                            ..
+                        },
+                    ..
+                } => match state {
+                    ElementState::Pressed => {
+                        println!("hey");
```
Comment (azriel91) [inline amethyst_ui/src/text.rs]: heya :wave:

**Comment by azriel91** [comment]
Yay, am glad you managed to fix it :bowing_man: (I became frustrated trying).

> It's a breaking change (well, I think it's not a big deal on its own), but it also can be quite obscure to users: the application will def crash if `InputBundle` is loaded after `UiBundle`

I think it's fine for a breaking change, an entry in `CHANGELOG.md` should be good enough to let people know (and we can call it out in a release blog post).

File: amethyst_ui/src/text.rs [hunk 2]
```diff
@@ -224,7 +224,6 @@ impl<'a> System<'a> for TextEditingMouseSystem {
                     ..
                 } => match state {
                     ElementState::Pressed => {
-                        println!("hey");
```
Comment (mvlabat) [inline amethyst_ui/src/text.rs]: hey :)

**Pull request merged**